ai_configurator.py

```diff
@@ -35,37 +16,39 @@ def __init__(self):
             pass
         
         # variables related to keeping track of conversation history and token usage
+
         self.conversation_history = {} # the history received from the client each time and to be used for truncation
         self.stringified_conversation_history = "" # to be passed in along with the query each time
         self.token_limit = 0 # the maximum number of tokens for a given provider
         self.used_tokens = 0 # keep track of the number of tokens used locally(user + bot combined)
         self.tokens_exceeded = False # flag to keep track of whether the number of used tokens has been exceeded
-        self.tokenizer = None # tokenizer for the model being used
         self.previous_thread = ""
 
-        # the provider and the model being used
-        self.active_provider = None
-        self.active_model = None
+        # keep track of current model and provider
+        self.active_provider_name = ""
+        self.active_model_name = ""
 
-    def set_model(self, model_name):
+    def set_model(self, provider: str, model_name: str) -> None:
         """Set the active AI provider based on user input."""
-        match_found = False
-        for model in self.available_models:
-            if model == model_name:
-                self.active_model = model
-                match_found = True
-                self.token_limit = self.config[model]["context_length"]
-                if "model_id" not in self.config[model]: # gpt/gemini model
-                    is_openai = model.startswith("gpt")
-                    self.active_provider = "openai" if is_openai else "gemini"
-                    self.tokenizer = tiktoken.get_encoding("cl100k_base") if is_openai else tokenization.get_tokenizer_for_model(model)
-                else: # ollama model
-                    self.active_provider = "ollama"
-                    self.tokenizer = Tokenizer.from_pretrained(self.config[model]["model_id"], auth_token=self.hf_key)
-                break
-        
-        if not match_found:
-            raise ValueError(f"Unsupported model or missing API key: {model_name}")
+        # we will only execute this whenever the provider and/or the model name is changed
+        if provider != self.active_provider_name or model_name != self.active_model_name:
+            try:
+                module_name = f"model-config.{provider}.{provider}-config"
+                module = importlib.import_module(module_name)
+                classes = inspect.getmembers(module, inspect.isclass)
+                classes = list(filter(lambda x: x[1].__module__ == module_name, classes))
```

Comment:
Having one config file is better than having numerous config files based on provider.